### PR TITLE
Fix datesort issue for sales reports

### PR DESCRIPTION
**Pull request by DurkoMatko**
**fix**: sorting by date in lists using tablesorter
          (https://mottie.github.io/tablesorter/docs/example-option-date-format.html)

**add**: list template can now take a parameter to define class for <th> tag

### CODE REVIEW

**Comment by DurkoMatko** [comment]
[Fix datesort issue for sales reports](https://trello.com/c/x0EFIwGL/454-fix-datesort-issue-for-sales-reports)

File: include/sales_list.php [hunk 1]
```diff
@@ -90,7 +90,7 @@
                     $data[$key]['people'] = db_value('SELECT COUNT(DISTINCT(p.id)) FROM people AS p, transactions AS t WHERE (p.id = t.people_id OR p.parent_id = t.people_id) AND t.transaction_date >= "'.$date.' 00:00" AND t.transaction_date <= "'.$date.' 23:59" AND t.product_id > 0 AND t.people_id > 0 AND camp_id = :campid', ['campid' => $_SESSION['camp']['id']]);
                     $totalvisitors += $data[$key]['people'];
                 }
-                addcolumn('text', 'Sales', 'salesdate');
+                addcolumn('text', 'Sales', 'salesdate', 'sorter-shortDate dateFormat-ddmmyyyy');    //tablesorter works with dates only if header has a proper class assigned - https://mottie.github.io/tablesorter/docs/example-option-date-format.html
```
Comment (HaGuesto) [inline include/sales_list.php]: Please change to 
```
addcolumn('text', 'Sales', 'salesdate', ['headerClass' => 'sorter-shortDate dateFormat-ddmmyyyy']);
```

File: library/lib/list.php [hunk 2]
```diff
@@ -609,11 +609,11 @@ function query_insert($str, $pos, $insert)
     return substr($str, 0, $pos).$insert.' '.substr($str, $pos);
 }
 
-function addcolumn($type, $label = false, $field = false, $array = [])
+function addcolumn($type, $label = false, $field = false, $headerClass = false, $array = [])
```
[the remark this answers is not in the document]
Comment (DurkoMatko) [reply library/lib/list.php]: oops srry my bad :D

**Comment by DurkoMatko** [comment]
@HaGuesto updated

**Pull request merged**